internal/charm/repository/charmhub.go

```diff
@@ -353,8 +365,38 @@ func transformRefreshResult(charmName string, refreshResult transport.RefreshRes
 	return corecharm.EssentialMetadata{Meta: chMeta, Config: chConfig, Manifest: chManifest}, nil
 }
 
-// DownloadCharm retrieves specified charm from the store and saves its
-// contents to the specified path.
+// Download retrieves a blob from the store and saves its contents to the
+// specified path.
+//
+// To get the contents of the blob, read the path on success.
+func (c *CharmHubRepository) Download(ctx context.Context, name string, requestedOrigin corecharm.Origin, path string) (corecharm.Origin, *charmhub.Digest, error) {
```

Comment:
This shouldn't be here and will be moved in future PR. The repository should be about data not about binary blobs, the same goes for the charmhub client.

Future problem.